lib/class-wp-rest-autosaves-controller.php

```diff
@@ -309,7 +309,7 @@ public function create_post_autosave( $post_data ) {
 
 			if ( ! $autosave_is_different ) {
 				wp_delete_post_revision( $old_autosave->ID );
-				return new WP_Error( 'rest_autosave_no_changes', __( 'There is nothing to save. The autosave and the post content are the same.', 'gutenberg' ) );
+				return new WP_Error( 'rest_autosave_no_changes', __( 'There is nothing to save. The autosave and the post content are the same.', 'gutenberg' ), array( 'status' => 200 ) );
```

Comment:
I think `400` is more appropriate here than `200`. We should return `400` from the REST API, and then have special handling in Gutenberg for a `400` from the autosaves endpoint.